### ISSUE TITLE
Introduce Macro.generate_arguments(int)

### ISSUE THREAD

**Issue by josevalim** [elixir-lang/elixir#5689]
It will receive an int >= 0 and return a list of variables that can be used as arguments. We use this pattern a couple places in Elixir but I can't recall quite where right now.

**Comment by felipesere** [comment]
This sounds interesting. Can you point me in the right direction? What problem would this solve?

**Comment by lexmag** [comment]
@felipesere sure, this function will return list of `{name, meta, context}` tuples (check `quote(do: qwe)` to see real example) that can be used in following way:
```elixir
args = Macro.generate_arguments(3)
def main(unquote_splicing(args)) do
  ...
end
```

**Comment by felipesere** [comment]
Thank you for the example of the base structure. Grepping for `quote(do: ...)` resulted in ~300 hits as it is used everywhere. 
Can you narrow down the `quote(do: ..)` reference? 😄

**Comment by felipesere** [comment]
I've created a first minimal test and implementation.

I wonder how to make sure the `def main(...) do end` example @lexmag works too...

Feedback welcome!

**Comment by craigp** [comment]
Urgh, I didn't see that someone else was already working on this. You can disregard and close my PR if you like, apologies @felipesere

**Comment by felipesere** [comment]
@craigp I carried over your doc and `generate_arguments/2` 😃

**Comment by whatyouhide** [comment]
Closed by #5699.

@josevalim if you want to keep this open to remember to start using the newly added function in the codebase, please feel free to reopen 💟
